osu.Game/Graphics/UserInterfaceV2/RoundedButton.cs

```diff
@@ -107,5 +108,7 @@ public bool MatchingFilter
         }
 
         public bool FilteringActive { get; set; }
+
+        public virtual LocalisableString TooltipText { get; set; }
```

Comment:
This entire change with making `TooltipText` virtual is quite dodgy because some derived classes override the getter but leave the setter intact. Previously the intent was that the tooltip in those classes was derived from internal data and as such is immutable outside the class. But now with this you can freely overwrite the `TooltipText` of a `ReadyButton` or `FavouriteButton` or whatever else which is improper design.

The easiest way out of this would probably be to drop the `virtual`, and update the places that override the getter so that instead of it they use the setter (which will require a few bindable subscriptions or something to provide the same effect).

Reply:
This might be a good place to use the `new` keyword. By declaring a `public new LocalisableString TooltipText` in `ReadyButton` and `FavouriteButton` we can remove the setter entirely which seems more appropriate for these classes which want to manage their tooltip text internally.

Reply:
Eh. I don't like shadowing because it makes things tricky anyway. Simple non-virtual non-shadowed public `get; set;` is best.

Reply:
I would agree with you that shadowing makes things tricky in most cases, but in this case using shadowing seems like the simplest solution. If we just considered `ReadyButton` and `FavouriteButton` it would be quite a simple change because they use only bindables for their tooltip. However we have to consider the tooltip of `ReadyButton` actually gets overriden by `PlaylistsReadyButton` and `MultiplayerReadyButton` which each use several non-bindable variables in determining their tooltip text.
I'd have to add new bindables in several classes and it would become a quite complicated chain of bindable updates.